plugins/aws/es/esHttpsOnly.js

```diff
@@ -62,5 +82,63 @@ module.exports = {
         }, function() {
             callback(null, results, source);
         });
+    },
+    remediate: function(config, cache, settings, resource, callback) {
+        var putCall = this.actions.remediate;
+        var pluginName = 'esHttpsOnly';
+        var domainNameArr = resource.split(':');
+        var domain = domainNameArr[domainNameArr.length - 1].split('/')[1];
+
+        // find the location of the domain needing to be remediated
+        var domainLocation = domainNameArr[3];
+
+        // add the location of the domain to the config
+        config.region = domainLocation;
+        var params = {};
+
+        // create the params necessary for the remediation
+        if (settings.input &&
+            settings.input.tlsSecurityPolicyforEs) {
+            params = {
+                DomainName: domain,
+                DomainEndpointOptions: {
+                    EnforceHTTPS: true,
+                    TLSSecurityPolicy: settings.input.tlsSecurityPolicyforEs
+                },
+            };
+        } else {
+            params = {
+                DomainName: domain,
+                DomainEndpointOptions: {
+                    EnforceHTTPS: true,
+                    TLSSecurityPolicy: 'Policy-Min-TLS-1-2-2019-07'
```

Comment:
Can we have this value in some config files. so that if some other plugin requires we can use the same and it will be a single place to change the value if needed.

Reply:
Yes, we can. But there are a lot of places where we are defining this kind of settings inline, so I think it's consistent this way. Having said, we can do an activity where we can add these settings to some enum file and use them from there. What do you think?